[Cui Shuyuan] iP

legionlegion opened the pull request:
### Project Duke by [Shuyuan](https://www.linkedin.com/in/cui-shuyuan/) 😉

> "I love CS2103T" - Shuyuan

- Functional CLI schedule manager
- Validates user input
- Quick to use for users who can type fast
- Future GUI to be implemented

**Supports the following features:**
1. todo tasks
2. deadline tasks with end date
3. event tasks with start and end date
4. list all previous tasks
5. mark/unmark tasks
6. delete tasks
7. find tasks

**Current progress**
- [x] CLI control
- [ ] GUI control

Here is how it all starts in the `Duke.java` file:
```java
    public static void main(String[] args) {
        new Duke().run();
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

legionlegion committed:
Level-0

legionlegion committed:
Level-1

legionlegion committed:
Level-2

legionlegion committed:
Level-3

legionlegion committed:
Level-4 with Automated Text UI testing

legionlegion committed:
Level-5

legionlegion committed:
Level-6

legionlegion committed:
Level-6

legionlegion committed:
Merge branch 'master' of https://github.com/legionlegion/ip

legionlegion committed:
Edit code to follow Java coding standard

legionlegion committed:
Add Level 7 features and coding standards

Changed

Add functionality to save and load tasks to/from './data/duke.txt' file

Changed formatting across all classes to follow coding standards

legionlegion committed:
Change relative file path and edit error message

legionlegion committed:
Merge branch 'branch-Level-7'

legionlegion committed:
Add Level 8. Date and Times features

legionlegion committed:
Merge branch 'branch-Level-8'

legionlegion committed:
Add OS-independent paths

legionlegion committed:
Implement A-MoreOOP and A-Packages

Refactored the codebase to align with Object-Oriented Programming
principles. Extracted related functionalities into distinct classes:
- Ui,
- Storage,
- Parser, and
- TaskList

Remove Chatbot class as it has been refactored into other classes

- Ui class handles all user interactions.
- Storage class manages data persistence, loading and saving tasks.
- Parser class interprets user commands.
- TaskList class manages the list of tasks, including addition
and deletion operations.

Add all files to duke package.

legionlegion committed:
Update tasks.txt path and delete Chatbot.java

legionlegion committed:
Merge branch 'A-MoreOOP'

legionlegion committed:
Merge branch 'add-gradle-support' of https://github.com/legionlegion/ip

legionlegion committed:
Add Gradle and JUnit tests

Use Gradle to automate some of the build tasks
of the project

Add JUnit tests to test the behavior of the code
for the classes Parser and TaskList

legionlegion committed:
Merge branch 'A-JUnit'

legionlegion committed:
Add JavaDocs to all files

legionlegion committed:
Merge branch 'A-JavaDoc'

legionlegion committed:
Edit coding standard

Amend previous JavaDoc

legionlegion committed:
Merge branch 'A-CodingStandard'

legionlegion committed:
Add find feature for Level-9

legionlegion committed:
Merge branch 'Level-9'

legionlegion committed:
Add functional GUI prototype

legionlegion committed:
Merge branch 'Level-10'

meenulekha commented on src/main/java/duke/CommandType.java:
```diff
@@ -0,0 +1,25 @@
+ * Provides a method to convert a string command into a corresponding
+ * CommandType.
+ */
+public enum CommandType {
```
Good use of enum!

meenulekha commented on src/main/java/duke/ErrorHandler.java:
```diff
@@ -0,0 +1,38 @@
+ * Duke application.
+ * This includes file not found errors, I/O errors, and number format errors.
+ */
+public class ErrorHandler {
```
Using an error handler class is a very efficient way of handling the errors, nice!

meenulekha commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,79 @@
+     * @param start       The start date of the event.
+     * @param end         The end date of the event.
+     */
+    Event(String description, String startToEnd, String input, LocalDate start, LocalDate end) {
```
I think you can specify if the constructor is public!

meenulekha commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,241 @@
+        Task newTask = null;
+        switch (command) {
+        case TODO:
+            if (parts[1].replaceAll("\\s", "").equals("")) {
```
Maybe you can use a variable before the switch statement to represent parts[1].replaceAll("\\s", "") in the if statements, it can help with readability!

myang2020 commented on src/main/java/duke/CommandType.java:
```diff
@@ -0,0 +1,25 @@
+ * CommandType.
+ */
+public enum CommandType {
+    TODO, DEADLINE, EVENT, MARK, UNMARK, DELETE, BYE, LIST, FIND, INVALID;
```
Great job following all caps convention for enums!

myang2020 commented on src/main/java/duke/DialogBox.java:
```diff
@@ -0,0 +1,48 @@
+import javafx.scene.Node;
+import javafx.scene.control.Label;
+import javafx.scene.image.ImageView;
+import javafx.scene.layout.HBox;
```
Great job listing out all imports to avoid import conflicts.

myang2020 reviewed (approved):
Could not find any violations of Java coding standard.

myang2020 commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,67 @@
+ * A deadline task includes a description, a due date, and the original input.
+ */
+public class Deadline extends Task {
+    private String by; // Formatted due date string
```
Perhaps use a more descriptive variable name such as "deadline"

myang2020 commented on src/main/java/duke/Todo.java:
```diff
@@ -0,0 +1,55 @@
+package duke;
```
Perhaps split classes into multiple packages other than just the single duke package.

kervyntan commented on src/main/java/duke/GUI.java:
Can consider using changing name of file to Gui.java!

kervyntan replied:
Other methods are well-named according to the standard, except maybe getResponse and handleUserInput, can afford to be a little more specific with those!

kervyntan commented on src/main/java/duke/Duke.java:
Nice note of the capitalized lettering for DATA_DIRECTORY!

kervyntan reviewed (commented):
Overall well done! Minor changes in terms of feedback

legionlegion committed:
Move all CLI output to GUI

legionlegion committed:
Add assert feature to Task modification functions

legionlegion committed:
Merge pull request #2 from legionlegion/branch-A-Assertions

Branch A-Assertions

legionlegion committed:
Refactor UI and parsing logic to adhere to SLAP

1. Abstracted `getResponse` into `getResponse` and `processCommand`
with multiple helper methods for command parsing
2. Abstracted `parse` function body into multiple helper methods
3. Abstracted `createTask` function body into multiple helper methods
for task creation
4. Transfer CLI logging logic into GUI display

legionlegion committed:
Merge pull request #3 from legionlegion/branch-A-CodeQuality

Refactor UI and parsing logic to adhere to SLAP

legionlegion committed:
Merge branch 'master' of https://github.com/legionlegion/ip

legionlegion committed:
Add B-DetectAnomalies function for events

- Add additional function to check event start time
is before end time

legionlegion committed:
Merge pull request #4 from legionlegion/B-DetectAnomalies

B detect anomalies

legionlegion committed:
Refactor existing code

legionlegion committed:
Merge master branch with master-refactor

legionlegion committed:
Update Ui.png and assert statements

legionlegion committed:
Update README.md

legionlegion committed:
Add user guide

legionlegion committed:
Merge branch 'master' of https://github.com/legionlegion/ip
